plugins/reolink/src/main.ts

```diff
@@ -1149,63 +1153,119 @@ class ReolinkProvider extends RtspProvider {
         ];
     }
 
+    getDevice(nativeId: string) {
+        if (nativeId.endsWith('-reolink-nvr')) {
+            let ret = this.nvrDevices.get(nativeId);
+            if (!ret) {
+                ret = this.createNVRDevice(nativeId);
+                if (ret)
+                    this.nvrDevices.set(nativeId, ret);
+            }
+
+            return ret;
+        } else {
+            return super.getDevice(nativeId);
+        }
+    }
+
     async createDevice(settings: DeviceCreatorSettings, nativeId?: string): Promise<string> {
-        const httpAddress = `${settings.ip}:${settings.httpPort || 80}`;
         let info: DeviceInformation = {};
 
         const skipValidate = settings.skipValidate?.toString() === 'true';
         const username = settings.username?.toString();
         const password = settings.password?.toString();
-        let doorbell: boolean = false;
-        let name: string = 'Reolink Camera';
-        let deviceInfo: DevInfo;
-        let ai;
-        let abilities;
-        const rtspChannel = parseInt(settings.rtspChannel?.toString()) || 0;
-        if (!skipValidate) {
-            const api = new ReolinkCameraClient(httpAddress, username, password, rtspChannel, this.console);
-            const apiWithToken = new ReolinkCameraClient(httpAddress, username, password, rtspChannel, this.console, true);
-            try {
-                await api.jpegSnapshot();
-            }
-            catch (e) {
-                this.console.error('Error adding Reolink camera', e);
-                throw e;
+        const ip = settings.ip?.toString();
+        const httpPort = settings.httpPort;
+        const rtspPort = settings.rtspPort;
+        const isNvr = settings.isNvr?.toString() === 'true';
+        const httpAddress = `${ip}:${httpPort || 80}`;
+
+        if (isNvr) {
```

Comment:
lets move this entire block into a `return createNvrDevice`. Then remove the else block so the original code doesn't need to be branched.